prov/shm/src/smr.h

```diff
@@ -78,6 +78,7 @@ int smr_fabric(struct fi_fabric_attr *attr, struct fid_fabric **fabric,
 struct smr_av {
 	struct util_av		util_av;
 	struct smr_map		*smr_map;
+	size_t			used;
```

Comment:
Instead of adding a new variable and tracking it, let's let the util_av handle this. If we set the max size when we create it, it will fail the ofi_av_insert_addr call and we can go through the eq write event path.
The way to do this would be to add a case when verifying the attr->count in smr_av.c:216 when an
```
else if (!attr->count)
    attr->count = SMR_MAX_PEERS;
```

Reply:
@aingerson  I thought about this. The problem I had is that ofi_av_insert_addr can still return us a valid index, when more peers than SMR_MAX_PEERS got inserted, even though we set the attr->count to SMR_MAX_PEERS. This is because util_av allocate a bufpool. If it is used up, it does grow, and can return a valid index therefore. The benchmark I ran ended up getting an aborted signal. 
So I had the error handling in this way to handle it sooner than later.

Reply:
if attr->count is non-zero, it will set the bufpool->max_count attribute which will not grow passed that number of entries so it should handle this case.

Reply:
Hmm, I saw we only set `pool_attr.chunk_cnt = av->count` before calling ofi_bufpool_create_attr().

Reply:
I thought it did but on second glance it looks like it doesn't set the bufpool max_count but just uses that to set the av->count.
https://github.com/ofiwg/libfabric/blob/master/prov/util/src/util_av.c#L446
I thought that the av->count, if set, could be assumed to be a hard limit but perhaps I'm wrong. @shefty can you clarify? If an app opens at AV with count n does the (n+1)th have to succeed or can it fail?
If it is a hard limit, then I think if it's set, the util_av can set the bufpool max_count to be the attr->count. If not, then I'm ok with keeping a variable to track this

Reply:
av->count is not a hard limit from the perspective of the API

Reply:
Then I assume keeping a variable smr_av->used is acceptable to both of you.

Reply:
I'm deferring to @aingerson for that change.